### PR TITLE
Tidy up the test infrastructure related to (lens) unit tests.

### PR DESCRIPTION
**Pull request by dhil**
"Demotes" the (lens) unit tests from being a separate package to being simply an executable.

Also removed `test/` directory which I supposedly pulled in 9 months ago.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -19,4 +19,5 @@ script:
   - make nc-release
   - make tests
   - ./run-tests db-only shredding
-  - ./run-unit-tests
+  - opam install oUnit
```
[the remark this answers is not in the document]
Comment (dhil) [reply .travis.yml]: Good point.

**Comment by frank-emrich** [comment]
I feel like the idea of  unit/unitTests.ml (now tests/lens/unitTests.ml) was to  have one place where we would register all (possibly non-lens) unit tests instead of essentially having such a file for every (future) group of unit tests. Maybe we should create a dedicated unit directory in tests as the root of the unit tests (i.e., the lens tests would live in tests/unit/lens).

**Comment by dhil** [comment]
> Maybe we should create a dedicated unit directory in tests as the root of the unit tests (i.e., the lens tests would live in tests/unit/lens).

Yes, that makes sense when/if we get other "unit tests" than lens.

**Comment by dhil** [comment]
@rudihorn I deleted a ton of trailing whitespace in lens*.ml files. You should configure your editor to display trailing whitespace (or automatically delete it).

**Pull request merged**